source/mir/random/ndvariable.d

```diff
@@ -279,6 +280,118 @@ nothrow @safe version(mir_random_test) unittest
     assert(fabs(x[0] + x[1] + x[2] - 1) < 1e-10);
 }
 
+/++
+Multinomial distribution.
++/
+struct MultinomialVariable(U:size_t, T)
+    if (isFloatingPoint!T)
+{
+    import mir.random.variable : binomialVar;
+
+    ///
+    enum isNdRandomVariable = true;
+    ///
+    alias Element = uint;
+
+    ///
+    const(T)[] probs;
+    size_t N;
+
+    /++
+    Params:
+        probs = probabilities of the multinomial distribution
+        N = Number of rolls
+    Constraints: `sum(probs[i]) <= 1`
+    +/
+    this()(size_t N, const(T)[] probs)
+    {
+        this.N = N;
+        this.probs = probs;
+    }
+
+    ///
+    pragma(inline, false)
+    void opCall(G)(scope ref G gen, scope U[] result)
+        if (isSaturatedRandomEngine!G)
+    {
+        auto K = this.probs.length;
+        double norm = 0.0;
+        double sum_p = 0.0;
+        alias rng = rne;
+
+        uint sum_n = 0;
+
+
+        /// Makes sure probabilities add up to one, by calculating a normalization term
+        foreach(k, p; this.probs)
+        {
+
+            norm += p;
+        }
+
+        foreach(k, p; this.probs)
+        {
+            if (p > 0.0)
+            {
+
+                auto rv = binomialVar!T(this.N - sum_n, p / (norm - sum_p));
+                result[k] = cast(uint)rv(rng);
```

Comment:
Should this be `result[k] = cast(uint)rv(gen);`?

Reply:
yes, fixed.